identifier/backends/ldap/ldap.go

```diff
@@ -561,6 +568,20 @@ func (b *LDAPIdentifierBackend) connect(parentCtx context.Context) (*ldap.Conn,
 	return l, nil
 }
 
+// connect establishes a global connection that is used for all requests except the user Bind to check the password in the Logon call.
```

Comment:
Please explain how is this thread safe in the comment. This function can be called inparallel, means that multiple threads get the same connection. I have my doubts that this is great since essentially now this uses a "single" LDAP connection for all parallel logins, allowing only one a the time via the internal locks in the connection implementation (assuming it is thread safe).

I would suggest to explore if it is possible to easily add a pool of LDAP connections with this. change.

Reply:
Using a single LDAP connection from multiple threads should actually work fine. The go-ldap module handles that just find. (We're using this in ownCloud Infinite Scale quite a bit). And LDAP servers should be able to reasonably deal with that as well.

I know that e.g. OpenLDAP  limits in how many LDAP Request from a single connection are processed in parallel,   but otoh there are also limits on how may operation of a single "user" are processed in parallel. I have some doubts that "normal" lico instance will be affected by those limits.

In the longer run you're probably right, and some connection pool mechanism could be helpful. I would suggest to first try without that an see how far it gets us.

Reply:
OK, fair enough :+1: